test/unit/token/erc20/effects/transferFrom.ts

```diff
@@ -0,0 +1,146 @@
+              .transferFrom(this.signers.carol.address, this.signers.bob.address, newTransferAmount),
+          ).to.be.revertedWith(Erc20Errors.TransferUnderflow);
+        });
+      });
+
+      context("when the token owner has enough balance", function () {
+        beforeEach(async function () {
+          await this.contracts.erc20.connect(this.signers.carol).mint(this.signers.carol.address, transferAmount);
+        });
+
+        it("transfers the requested amount", async function () {
+          const tokenOwnerOldBalance: BigNumber = await this.contracts.erc20.balanceOf(this.signers.carol.address);
+          const recipientOldBalance: BigNumber = await this.contracts.erc20.balanceOf(this.signers.bob.address);
+
+          await this.contracts.erc20
+            .connect(this.signers.alice)
+            .transferFrom(this.signers.carol.address, this.signers.bob.address, transferAmount);
+
+          const tokenOwnerNewBalance: BigNumber = await this.contracts.erc20.balanceOf(this.signers.carol.address);
+          const recipientNewBalance: BigNumber = await this.contracts.erc20.balanceOf(this.signers.bob.address);
+
+          expect(tokenOwnerNewBalance).to.equal(tokenOwnerOldBalance.sub(transferAmount));
+          expect(recipientNewBalance).to.equal(recipientOldBalance.add(transferAmount));
+        });
+
+        it("decreases the spender allowance", async function () {
+          const oldAllowanceAmount: BigNumber = await this.contracts.erc20
+            .connect(this.signers.alice)
+            .allowance(this.signers.carol.address, this.signers.alice.address);
+          await this.contracts.erc20
+            .connect(this.signers.alice)
+            .transferFrom(this.signers.carol.address, this.signers.bob.address, transferAmount);
+          const newAllowanceAmount: BigNumber = await this.contracts.erc20
+            .connect(this.signers.alice)
+            .allowance(this.signers.carol.address, this.signers.alice.address);
+          expect(newAllowanceAmount).to.be.equal(oldAllowanceAmount.sub(transferAmount));
+        });
+
+        it("emits a transfer event", async function () {
+          await expect(
+            this.contracts.erc20
+              .connect(this.signers.alice)
+              .transferFrom(this.signers.carol.address, this.signers.bob.address, transferAmount),
+          )
+            .to.emit(this.contracts.erc20, "Transfer")
+            .withArgs(this.signers.carol.address, this.signers.bob.address, transferAmount);
+        });
+
+        it("emits a approval event", async function () {
+          await expect(
+            this.contracts.erc20
+              .connect(this.signers.alice)
+              .transferFrom(this.signers.carol.address, this.signers.bob.address, transferAmount),
+          )
+            .to.emit(this.contracts.erc20, "Approval")
+            .withArgs(this.signers.carol.address, this.signers.alice.address, transferAmount);
```

Comment:
More specifically:

```ts
it("emits an Approval event", async function () {
  await expect(
    this.contracts.erc20
      .connect(this.signers.alice)
      .transferFrom(this.signers.carol.address, this.signers.bob.address, transferAmount),
  )
    .to.emit(this.contracts.erc20, "Approval")
    .withArgs(this.signers.carol.address, this.signers.alice.address, Zero);
});
```

Notice the `Zero` instead of `transferAmount`.

Reply:
Right, my bad.